RFCs/0019-Disable-Appdomain-While-Running-Tests.md

```diff
@@ -0,0 +1,28 @@
+# 0019 Disable Appdomain while Running Tests
+
+## Summary
+The RFC outlines why
+
+1. Test adapters should honour disable app domain setting
+
+2. Running of tests with disable appdomain is important for resiliency 
+
+## Motivation
+In past we have seen customer hitting issue with Appdomain.Unload. There are two main issues with Appdoamin.Unload call
+
+1. Hang in Appdoamin.Unload (Tracking issue https://github.com/Microsoft/testfx/issues/225)
+
+2. AppDomain.Unload call crashes the process even if you have an exception handler in code. This is due to a CLR and OS bug which ends up ignoring the exception handler in code and terminates the process.
+
+## Proposed changes
+
+Proposed guidelines are for customers and test adapters who wants to avoid tese issues.
+
+**Adapters**
+1. Test Adapter should honour ```<DisableAppDomain>``` setting inside RunConfiguration node of runsettings. Check https://github.com/Microsoft/vstest-docs/blob/master/docs/configure.md for information on this setting. This will ensure that adapters dont create AppDomain at all to run tests
+
+**Test platform**
+
+1. Change in test platform to merge the app.config for a test assembly when ```<DisableAppDomain>``` is set. This is to ensure test's app.config is honoured while running tests
+
+2. Make sure when ```<DisableAppDomain>``` is set, test platform should make ```<MaxCpuCount>``` to 1 (or use MaxCpuCount specified by the user if there). This is to ensure each test source have isolation. In case MaxCPUCount is set to 1 there will be one test host process created for each source dll.
```

Comment:
`MaxCpuCount` not relevant.

Reply:
it is. this will ensure that people know what the behaviour with disabledappdomain

---
In reply to: [177700474](https://github.com/Microsoft/vstest-docs/pull/123#discussion_r177700474) [](ancestors = 177700474)

Reply:
done